src/assets/Keycloak/oCIS-with-Keycloak.md

```diff
@@ -0,0 +1,195 @@
+
+![oCIS](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-1.png)
+
+2. Go to Keycloak url: `https://host.docker.internal:8443/`
+
+![keycloak](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-2.png)
+
+3. Create new realm: `oCIS`
+
+![create realm](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-1.png)
+![create realm with name `oCIS`](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-2.png)
+> Note: the realm name should be the name provided in the oCIS environment variable: `OCIS_OIDC_ISSUER: https://host.docker.internal:8443/realms/oCIS`
+
+4. Add a new client:
+
+![client registration](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-1.png)
+
+- Client ID: `web`
+
+![add client](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-2.png)
+
+- Root URL: `https://host.docker.internal:9200`
+
+![add root url](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-3.png)
+
+5. Add realm roles:
+
+- Role name: `ocisAdmin`
+
+![add realm role](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-role.png)
+
+> Note: oCIS has defined roles name that has been documented under [Automatic Quota Assignments](https://owncloud.dev/services/proxy/#automatic-role-assignments). So, we can't create the role with a random name.
+
+6. Create a new user:
+
+- With `username: admin` and other information:
+
+![create user](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-1.png)
+![add user information](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-2.png)
+
+- Create password
+
+![create password](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-3.png)
+![setup password for user](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-4.png)
+
+- Role Mapping: Assign `ocisAdmin` role
+
+![assign role](/src/assets/Keycloak/images/oCIS-Keycloak/assign-role-to-user-1.png)
+
+![assign role to user](/src/assets/Keycloak/images/oCIS-Keycloak/asign-role-to-user-2.png)
+
+7. Update `roles` Client scope: `Client Scopes` -> `roles`
+
+- `Mappers` -> `realm roles` -> `Token Claim Name=roles`
+![update-roles](/src/assets/Keycloak/images/oCIS-Keycloak/update-roles.png)
+> Note: oCIS default role_claim is 'roles'
+
+8. Go to url `https://host.docker.internal:9200/`, reload if already opened
+![login to oCIS with keycloak](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-login.png)
+Enter `username: admin` and `password: admin` to log into oCIS
+
+### Setting Up oCIS With Keycloak Using JSON files
+Keycloak provides an [Importing and Exporting Realms](https://www.keycloak.org/server/importExport) feature. If we have JSON configuration files then we can avoid setting services in Keycloak.
+oCIS provides documentation to set up [oCIS with Keycloak](https://owncloud.dev/ocis/deployment/ocis_keycloak/) which uses JSON configuration. oCIS provides source code to deploy
+[oCIS-with-Keycloak](https://github.com/owncloud/ocis/tree/master/deployments/examples/ocis_keycloak).
```

Comment:
Hyphens are mainly used in English for compound nouns, but it's quite uncommon to use them for a term such as "oCIS with Keycloak". See also how this is spelled in the official documentation by ownCloud: https://owncloud.dev/ocis/deployment/ocis_keycloak/